beacon-chain/sync/pending_attestations_queue.go

```diff
@@ -43,6 +45,8 @@ func (s *Service) processPendingAtts(ctx context.Context) error {
 	// be deleted from the queue if invalid (ie. getting staled from falling too many slots behind).
 	s.validatePendingAtts(ctx, s.chain.CurrentSlot())
 
+	s.pendingAttsLock.RLock()
```

Comment:
I think you're going to need the Lock here since you are deleting values from the map